Src/RE2.Managed/NativeMethods.cs

```diff
@@ -76,6 +76,30 @@ public static unsafe int Matches(int regexIndex, String8Interop text, int fromTe
                 : NativeMethodsX86.Matches(regexIndex, text, fromTextIndex, matches, matchesLength, timeoutMilliseconds);
         }
 
+        public static unsafe void GetRegexSetup(StringUtf8 pattern, ulong* numCapturingGroups, ulong* numNamedCapturingGroups, ulong* numSubmatches, ulong* groupNameHeadersBufferSize, ulong* groupNamesBufferSize, ulong* submatchesBufferSize)
```

Comment:
This should probably be integrated into the existing build-and-cache regex approach.